src/core/DropdownMenu/index.tsx

```diff
@@ -0,0 +1,167 @@
+  StyleProps,
+} from "./style";
+
+// (thuang): This requires option to have a `name` property.
+export interface DefaultDropdownMenuOption {
+  name: string;
+  section?: string;
+  details?: string;
+  count?: string;
+}
+
+interface ExtraProps extends StyleProps {
+  renderInput?: (params: AutocompleteRenderInputParams) => React.ReactNode;
+  onInputChange?: (event: React.SyntheticEvent) => void;
+  InputBaseProps?: Partial<InputSearchProps>;
+}
+
+type CustomAutocompleteProps<
+  T,
+  Multiple extends boolean | undefined = undefined,
+  DisableClearable extends boolean | undefined = undefined,
+  FreeSolo extends boolean | undefined = undefined
+> = Omit<
+  AutocompleteProps<T, Multiple, DisableClearable, FreeSolo>,
+  "renderInput"
+>;
+
+export type DropdownMenuProps<
+  T,
+  Multiple extends boolean | undefined = undefined,
+  DisableClearable extends boolean | undefined = undefined,
+  FreeSolo extends boolean | undefined = undefined
+> = CustomAutocompleteProps<T, Multiple, DisableClearable, FreeSolo> &
+  ExtraProps;
+
+export default function DropdownMenu<
+  T extends DefaultDropdownMenuOption,
+  Multiple extends boolean | undefined = undefined,
+  DisableClearable extends boolean | undefined = undefined,
+  FreeSolo extends boolean | undefined = undefined
+>(
+  props: DropdownMenuProps<T, Multiple, DisableClearable, FreeSolo>
+): JSX.Element {
+  const {
+    multiple = false,
+    getOptionLabel = defaultGetOptionLabel,
+    getOptionSelected = defaultGetOptionSelected,
+    renderTags = defaultRenderTags,
+    renderOption = defaultRenderOption,
+    disableCloseOnSelect = multiple,
+    noOptionsText = "No options",
+    search = false,
+    onInputChange = noop,
+    InputBaseProps = {},
+  } = props;
+
+  return (
+    <div>
+      <StyledAutocomplete
+        clearOnBlur={false}
+        disableCloseOnSelect={disableCloseOnSelect}
+        disablePortal
+        renderTags={renderTags}
+        noOptionsText={noOptionsText}
+        renderOption={renderOption}
+        getOptionLabel={getOptionLabel}
+        getOptionSelected={getOptionSelected}
+        renderInput={(params) => (
+          <InputBaseWrapper search={search}>
+            <StyledMenuInputSearch
+              id="location-search"
+              label="Search for a location"
+              placeholder="Search"
+              ref={params.InputProps.ref}
+              search={search}
+              onChange={onInputChange}
+              autoFocus
+              InputProps={{
+                /**
+                 * (thuang): Works with css caret-color: "transparent" to hide
+                 * mobile keyboard
+                 */
+                inputMode: search ? "text" : "none",
+                /**
+                 * (mmoore): passing only the ref along to InputProps to prevent
+                 * default MUI arrow from rendering in search input.
+                 * renderInput strips InputProps, so we explictly pass end adornment here
```

Comment:
```suggestion
                 * renderInput strips InputProps, so we explicitly pass end adornment here
```

Install https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker for spell checking if needed!

Reply:
Fixed 👍🏻 
Thank you for the tip.